### ISSUE TITLE
 Pre-submission enquiry

### ISSUE THREAD

**Issue by HPicatto** [openjournals/joss#1339]
I am currently preparing to submit a paper to the Journal of Open Source Software and intend to publish the accompanying code. The codebase is housed in a private GitHub repository. Due to confidentiality constraints, not all content in the repository can be made public at the time of the paper's publication.
To comply with JOSS's open-source requirements, I am considering creating a parallel public repository that includes only the components releasable at this stage. However, I am concerned that this approach might result in the loss of the original commit history and the contributions' authorship records.
Could you please advise on the best practices or recommend strategies for managing this situation?

**Comment by logological** [comment]
Are the individual components that you can publically release under an OSI-approved licence usable and useful without the ones that you will be holding back?  If not, then the software may not meet JOSS's core "obvious research application" requirement.

Regarding the authorship records, AFAIK there's no requirement that these take the form of a Git commit log.  You could instead credit the authors in comments in the individual source files, and/or in a dedicated text file listing the contributors.

**Comment by sneakers-the-rat** [comment]
not commenting on scope, but you could fork the repo and use [git-filter-repo](https://github.com/newren/git-filter-repo) to remove the stuff you need to keep private, preserving the rest of the repo history

**Comment by geoHeil** [comment]
@sneakers-the-rat do you have any experience with https://github.com/open-condo-software/gitexporter ?

**Comment by sneakers-the-rat** [comment]
no, but it seems pretty much like a tool to do what i'm thinking of :). make a fork/copy the repo and modify the remotes. `git-filter-repo` to remove private files and their histories from the repo and rebuild its history. preserve remaining history and etc. if i wanted to keep them linked then i would probably just automate merges/cherrypicks excluding some directories. if the history is less important in the secondary repo, you could just continually filter, rebuild, and force push to ensure you don't get pollution between the repos.

**Comment by brettmartin4** [comment]
> Are the individual components that you can publically release under an OSI-approved licence usable and useful without the ones that you will be holding back? If not, then the software may not meet JOSS's core "obvious research application" requirement.

But if so, is there wiggle room? I'm curious as to whether the requirement for a repo to be public-facing for at least 6 months is waiverable if an author is able to produce substantial proof of work history.

**Comment by warrickball** [comment]
Hi @brettmartin4, note that we updated our [scope and significance requirements](https://joss.readthedocs.io/en/latest/submitting.html#scope-and-significance) at the beginning of 2026 (see [the announcement](https://blog.joss.theoj.org/2026/01/preparing-joss-for-a-generative-ai-future)) and they now specifically say:

> Projects developed privately are not eligible until there is a public record of open development: at least six months of public history prior to submission, ...

We've been treating that fairly strictly since these came into effect.

Separately though, I noticed the prospective paper mentioned here has [been reviewed](https://github.com/openjournals/joss-reviews/issues/7695) and published, so I'm going to close this.
